tests/test_utils.py

```diff
@@ -142,3 +152,38 @@ def captured_output() -> Generator[Tuple[TextIO, TextIO], None, None]:
         yield sys.stdout, sys.stderr
     finally:
         sys.stdout, sys.stderr = old_out, old_err
+
+
+def gpu_test(gpu_count: int = 1):
+    """
+    Annotation for GPU tests, skipping the test if the
+    required amount of GPU is not available
+    """
+    message = f"Not enough GPUs to run the test: requires {gpu_count}"
+    local_gpu_count: int = torch.cuda.device_count()
+    return pytest.mark.skipif(local_gpu_count < gpu_count, reason=message)
+
+
+def get_loss_values_from_metric_logger(log_file_path: str) -> Dict[str, float]:
+    """
+    Given an output directory containing metric logger .txt file,
+    parse the .txt and return a list of losses from each logged iteration.
+    """
+    with open(log_file_path, "r") as f:
+        logs = f.read()
+    losses = [float(x) for x in re.findall(r"loss:(\d+\.\d+)", logs)]
+    return losses
+
+
+def gen_log_file_name(tmpdir, suffix: Optional[str] = None) -> str:
```

Comment:
I've come to appreciate how useful pathlib is for all of this. I think you can simplify this a lot of if you convert tmpdir into a ```Path``` object (I forgot if it's already a ```Path```). In that cass this begins:

```
filename = Path.joinpath(tmpdir, suffix).with_suffix('txt')
```

Reply:
Sorry, actually gonna leave this as-is. The point here is to generate a unique and deterministic filename per test case here. So I am actually deliberately taking tmpdir and removing all the `/`s to do this. In that case I think Path actually makes things look a bit more complicated, since I am doing some pretty non-Pathy operations